shared/helpers/puppeteer.js

```diff
@@ -111,20 +117,25 @@ async function renderAndInitialise(page, componentName, options) {
   // gather and `return` the values we need from inside the browser, and throw
   // them when back in Jest (to keep them triggering a Promise rejection)
   const error = await page.evaluate(
-    async (exportName, options) => {
-      const $module = document.querySelector('[data-module]')
-
+    async (selector, exportName, config) => {
       const namespace = await import('govuk-frontend')
 
+      // Find all matching modules
+      const $modules = document.querySelectorAll(selector)
+
       try {
-        /* eslint-disable-next-line no-new */
-        new namespace[exportName]($module, options.config)
+        // Loop and initialise all $modules or use default
+        // selector `null` return value when none found
+        ;($modules.length ? $modules : [null]).forEach(
```

Comment:
Is this semi-colon doing something fancy?

Reply:
Yeah caught me out too

It's a Prettier protection against those parentheses  `()` becoming a function call
https://prettier.io/docs/en/rationale#semicolons

[ESLint does it too](https://eslint.org/docs/latest/rules/semi) but not nearly as much